assets/js/modules/analytics-4/datastore/webdatastreams.js

```diff
@@ -322,6 +322,50 @@ const baseSelectors = {
 	getWebDataStreamsBatch( state, propertyIDs ) {
 		return pick( state.webdatastreams, propertyIDs );
 	},
+
+	/**
+	 * Gets the matched measurement IDs for the given properties.
+	 *
+	 * @since 1.88.0
+	 *
+	 * @param {Object}         state      Data store's state.
+	 * @param {Array.<Object>} properties GA4 properties array of objects.
+	 * @return {(Object|null)} Matched property ID as the key and measurement ID as the value; otherwise, an empty object. Null if no properties are provided.
+	 */
+	getMatchedMeasurementIDByPropertyID: createRegistrySelector(
```

Comment:
As this returns an object which can contain multiple properties/measurement IDs, we should reflect that in the name. Maybe just pluralising would suffice?

```suggestion
	getMatchedMeasurementIDsByPropertyIDs: createRegistrySelector(
```